esp-hal/build.rs

```diff
@@ -69,19 +69,51 @@ fn main() -> Result<(), Box<dyn Error>> {
     // Define all necessary configuration symbols for the configured device:
     config.define_symbols();
 
-    #[allow(unused_mut)]
-    let mut config_symbols = config.all().collect::<Vec<_>>();
-    #[cfg(feature = "flip-link")]
-    config_symbols.push("flip-link");
-
     // Place all linker scripts in `OUT_DIR`, and instruct Cargo how to find these
     // files:
     let out = PathBuf::from(env::var_os("OUT_DIR").unwrap());
     println!("cargo:rustc-link-search={}", out.display());
 
+    // emit config
+    let cfg = generate_config(
+        "esp_hal",
+        &[
+            (
+                "place-spi-driver-in-ram",
+                Value::Bool(false),
+                "Places the SPI driver in RAM for better performance",
+            ),
+            (
+                "spi-address-workaround",
+                Value::Bool(true),
+                "(ESP32 only) Enables a workaround for the issue where SPI in half-duplex mode incorrectly transmits the address on a single line if the data buffer is empty.",
+            ),
+            (
+                "place-more-rodata-in-ram",
```

Comment:
```suggestion
                "place-interrupt-switch-tables-in-ram",
```

bikeshedding welcome, but something more descriptive than "more" would be better here.

Reply:
agree if we can make it specific to the interrupt handling mechanism